client_test.go

```diff
@@ -242,7 +242,7 @@ func TestBasic_SMTPError(t *testing.T) {
 func TestClient_TooLongLine(t *testing.T) {
 	faultyServer := []string{
 		"220 mx.google.com at your service\r\n",
-		"220 mx.google.com at your service\r\n",
+		"250 mx.google.com at your service\r\n",
```

Comment:
Is this change intentional?

Reply:
Yes, this test started failing as a side effect of the change, but I think the test setup was wrong.

The first `220 mx.google.com at your service\r\n` would be consumed by `greet`, the second `220 mx.google.com at your service\r\n` would be consumed by `ehlo` which interprets anything other than 250 as an error and would fallback to using `helo`, which would capture the long line. 

With the changes I introduced in this PR, the `220` code won't trigger the `helo` and the long line is never consumed.

Does that make sense? My understanding is that the change is only breaking test assumptions.

Reply:
Ah yes indeed, 220 is not a valid response to EHLO or HELO. Good catch! The last line also has the same issue.

The message after the SMTP code is still confusing though: it looks like a greeting when it's an OK response.

I've fixed these issues in 8fc2197231c8.